### PR TITLE
feat: Implement 'Search Repos' Functionality on Stats Page

### PR DESCRIPTION
**Pull request by luckyklyist**
Fixes : #59 

I have added a text filter with an input component from DaisyUI. After entering text in the search input, it filters the repositories from the already fetched repositories. Then, it displays the filtered results on the page.

https://github.com/Balastrong/github-stats/assets/35479077/26bf58e5-d04a-4ff0-b9af-c07e7508d527

### CODE REVIEW

File: src/pages/stats/[login].tsx [hunk 1]
```diff
@@ -22,6 +22,27 @@ export default function Stats() {
...
+    const filterRepos = repositories.filter((repoData) =>
+      repoData.repository.name.toLowerCase().includes(query)
+    );
+
+    // checking if there is no repositories
+    if (filterRepos.length === 0) {
+      return [];
+    }
+
+    return filterRepos;
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:44]: Do we need to manually check if there are no repos?

`return repositories.filter(...)` should be enough

**Review by Balastrong — COMMENTED** [review]
Thank you for the PR, really well done!

I added only a question :)

File: src/pages/stats/[login].tsx (hunk 1 above)
Comment (luckyklyist) [reply src/pages/stats/[login].tsx]: Yeah, there is no need for that conditional statement. Thanks for the review. Just updated the code

**Comment by Balastrong** [comment]
LGTM, thank you! 🚀

**Comment by Balastrong** [comment]
@all-contributors please add @luckyklyist for code

**Pull request merged**